### ISSUE TITLE
Generator.cs TypeNameConflict

### ISSUE THREAD

**Issue by J-kit** [Splamy/TS3AudioBot#39]
bei `System.Reflection.Emit.ModuleBuilder.CheckTypeNameConflict(String strTypeName, Type enclosingType)`
Occurs when starting multible bots
```
  Doppelter Typenname in einer Assembly. - Duplicate type name in an assembly.
 bei System.Reflection.Emit.AssemblyBuilderData.CheckTypeNameConflict(String strTypeName, TypeBuilder enclosingType)
   bei System.Reflection.Emit.TypeBuilder.Init(String fullname, TypeAttributes attr, Type parent, Type[] interfaces, ModuleBuilder module, PackingSize iPackingSize, Int32 iTypeSize, TypeBuilder enclosingType)
   bei System.Reflection.Emit.ModuleBuilder.DefineType(String name, TypeAttributes attr, Type parent)
   bei TS3Client.Messages.Generator.Generate(Type backingInterface, Boolean notifyProp) in F:\File\Dokumente\Visual Studio 2015\Projects\TS3\3\TS3AudioBot-develop\TS3Client\Messages\Generator.cs:Zeile 66.
   bei TS3Client.Messages.Generator.Activate(Type backingInterface, Boolean notifyProp) in F:\File\Dokumente\Visual Studio 2015\Projects\TS3\3\TS3AudioBot-develop\TS3Client\Messages\Generator.cs:Zeile 51.
   bei TS3Client.Messages.Generator.ActivateNotification(Type t) in F:\File\Dokumente\Visual Studio 2015\Projects\TS3\3\TS3AudioBot-develop\TS3Client\Messages\Generator.cs:Zeile 39.
   bei TS3Client.CommandDeserializer.<>c__DisplayClass4_0.<GenerateNotification>b__0(String msg) in F:\File\Dokumente\Visual Studio 2015\Projects\TS3\3\TS3AudioBot-develop\TS3Client\CommandDeserializer.cs:Zeile 106.
   bei System.Linq.Enumerable.WhereSelectArrayIterator`2.MoveNext()
   bei System.Linq.Enumerable.FirstOrDefault[TSource](IEnumerable`1 source)
   bei TS3Client.Ts3BaseClient.<>c__DisplayClass46_0.<InvokeEvent>b__3() in F:\File\Dokumente\Visual Studio 2015\Projects\TS3\3\TS3AudioBot-develop\TS3Client\Ts3BaseClient.cs:Zeile 177.
   bei TS3Client.DoubleThreadEventDispatcher.EnterEventLoop() in F:\File\Dokumente\Visual Studio 2015\Projects\TS3\3\TS3AudioBot-develop\TS3Client\EventDispatcher.cs:Zeile 75.
   bei TS3Client.Ts3BaseClient.EnterEventLoop() in F:\File\Dokumente\Visual Studio 2015\Projects\TS3\3\TS3AudioBot-develop\TS3Client\Ts3BaseClient.cs:Zeile 113.
   bei TS3Client.Full.Ts3FullClient.<StartLoopThread>b__10_0() in F:\File\Dokumente\Visual Studio 2015\Projects\TS3\3\TS3AudioBot-develop\TS3Client\Full\Ts3FullClient.cs:Zeile 39.
```

**Comment by J-kit** [comment]
Ok, just found out that locking the object solves the problem.

**Comment by Splamy** [comment]
Is this a race condition problem, or does this error happen consistently?
Also would be nice to know where you locked what.

**Comment by J-kit** [comment]
It happens all the time when starting the Ts3Full simultaniously
Generator.cs ~line 47/object activate
https://gist.github.com/J-kit/294088268545f6ae039849bbec2d3d0e

Since then i never got that error again.

**Comment by Splamy** [comment]
Fixed with 6137e2413ef851125c02a2d7778d03863ccf8193
